node_modules/oae-zencoder/lib/processor.js

```diff
@@ -0,0 +1,130 @@
+
+var _ = require('underscore');
+var fs = require('fs');
+var request = require('request');
+var util = require('util');
+
+var ContentUtil = require('oae-content/lib/internal/util');
+var ContentConstants = require('oae-content/lib/constants').ContentConstants;
+var Context = require('oae-context').Context;
+var log = require('oae-logger').logger('oae-zencoder');
+var PreviewConstants = require('oae-preview-processor/lib/constants');
+
+var ZencoderDAO = require('./internal/dao');
+var ZencoderUtil = require('./util');
+
+/**
+ * @borrows Interface.test as Videos.test
+ */
+var test = module.exports.test = function(ctx, content, callback) {
+    if (content.resourceSubType === 'file') {
+        if (_isVideo(ctx.revision.mime) || _isAudio(ctx.revision.mime)) {
+            return callback(null, 30);
+        }
+    }
+
+    return callback(null, -1);
+};
+
+/**
+ * @borrows Interface.generatePreviews as Videos.generatePreviews
+ */
+var generatePreviews = module.exports.generatePreviews = function(ctx, content, callback) {
+    var revision = ctx.revision;
+
+    if (revision.previews && revision.previews.video) {
+        return callback();
+    }
+
+    log().info('Generating previews for video file: %s', content.id);
+    var tenant = content.tenant;
+
+    var downloadPath = ContentUtil.getSignedDownloadUrl(new Context(tenant), revision.uri, 6000, 6000);
+
+    // Create a Zencoder job
+    ZencoderUtil.job(tenant.alias, downloadPath, content.id, ctx.revisionId, _isVideo(revision.mime), function(err, res, body) {
+        if (_checkError(err, 'Error POSTing the video to Zencoder for encoding', res, body, callback)) {
+            return callback(err);
+        }
+
+        var output = body.outputs[0];
+
+        ZencoderDAO.saveContentRevisionId(output.id.toString(), content.id, ctx.revisionId, function(err) {
+            if (err) {
+                return callback(err);
+            }
+
+            ctx.setStatus(ContentConstants.previews.PENDING, function(err) {
+                if (err) {
+                    return callback(err);
+                }
+
+                return callback(null, false, true);
+            });
+        });
+    });
+};
+
+/**
+ * Determine whether or not the given mimeType is that of a video
+ *
+ * @param  {String}     mimeType        The mimetype to test
+ * @return {Boolean}                    `true` or `false`, indicating whether or not the given mime type indicates a video file
+ * @api private
+ */
+var _isVideo = function(mimeType) {
+    return (PreviewConstants.TYPES.VIDEO.indexOf(mimeType) !== -1);
+};
+
+/**
+ * Determine whether or not the given mimeType is that of an audio
+ *
+ * @param  {String}     mimeType        The mimetype to test
+ * @return {Boolean}                    `true` or `false`, indicating whether or not the given mime type indicates an audio file
+ * @api private
+ */
+var _isAudio = function(mimeType) {
+    return (PreviewConstants.TYPES.AUDIO.indexOf(mimeType) !== -1);
+};
```

Comment:
Rather than have `ZencoderUtil.isVideo` etc... like I mentioned before, since we already have some recognition of audio/video in the core preview constants, we could even have `PreviewUtil.isVideo(mimeType)` etc... which then could be re-used for other audio/video preview modules.